### DISCUSSION TITLE
Any likelihood of visual elements

### DISCUSSION THREAD

**Discussion by jjbourne** [Live-Charts/LiveCharts2#543] [General]
Hi beto,

Congrats on your progress to wards v2, I've started integrating it into a wpf app and it is looking very slick.

One usecase I have is to be able to drop vertical markers on the line chart - I will then display interpolated values for each series in a table somewhere on the screen.  I anticipated that maybe this could be achieved using something like the 'visual elements' from v1 of the library.  Does this functionality exist or is there something like it on the roadmap? 

Thanks,
Jon

**Comment by beto-rodriguez** [comment]
Hi and Thanks!

It is not implemented yet, but sure it will be!

**Reply by jjbourne** [reply]
That's great, thanks.

In the meantime, I have create something useable for me by making customised Section and Geometry classes (I've called it VerticalMarkerSection/VerticalMarkerGeometry). It is a start, but limited in some ways. I have implemented the OnDraw method and created a vertical line at the point the user places the cursor. This is great, it shows on the graph and moves around correctly with Pan/Zoom.

The part I'm not able to do is:
  - add a triangle shape at the top/bottom of the line that always shows, regardless of zoom level
  - add custom text (e.g. 'C1', 'C2' etc) to the section

I can draw the triangle shape, but zooming in causes it to disappear off the visible part of the chart. If you have any suggestions or I'm missing something obvious please let me know.

Thanks,
Jon
